tests/test_context.py

```diff
@@ -71,17 +73,43 @@ def test_context_items():
     assert set(ctx.items()) == expected_items
 
 
+def test_update_with_context_key_collition():
```

Comment:
```suggestion
def test_context_update_with_key_collision():
```

Same with the other guy (`def test_context_update_duplicate_with_replace`)